kafl_fuzzer/common/config/settings.py

```diff
@@ -0,0 +1,210 @@
+        str(Path(appdirs.site_config_dir) / SETTINGS_FILENAME),
+        # $HOME/.config/kAFL
+        str(Path(appdirs.user_config_dir) / SETTINGS_FILENAME),
+        # local
+        "kafl.yaml",
+    ]
+    # env var KAFL_CONFIG_FILE if present
+    env_config_file = os.environ.get('KAFL_CONFIG_FILE', None)
+    if env_config_file is not None:
+        # check if file exists and raise an error if not to warn user
+        if not Path(env_config_file).is_file():
+            raise ValidationError(f"KAFL_CONFIG_FILE: {env_config_file} not found")
+        settings_files.append(env_config_file)
+    return settings_files
+
+appdirs = AppDirs(APPNAME)
+settings = Dynaconf(
+    envvar_prefix=APPNAME.upper(),
+    settings_files=app_settings_files()
+)
+
+# validator condition funcs
+# TODO: should be used to validate --afl-skip-range
+# def parse_ignore_range(string: str) -> bool:
+#     m = re.match(r"(\d+)(?:-(\d+))?$", string)
+#     if not m:
+#         raise argparse.ArgumentTypeError("'" + string + "' is not a range of number.")
+#     start = min(int(m.group(1)), int(m.group(2)))
+#     end = max(int(m.group(1)), int(m.group(2))) or start
+#     if end > (128 << 10):
+#         raise argparse.ArgumentTypeError("Value out of range (max 128KB).")
+
+#     if start == 0 and end == (128 << 10):
+#         raise argparse.ArgumentTypeError("Invalid range specified.")
+#     return list([start, end])
+
+def cast_ip_range_to_list(parameter: Any) -> Optional[List[int]]:
+    """Checks that a given IP range string is valid and returns a List of that range"""
+    if parameter is None:
+        return None
+    m = re.match(r"([(0-9abcdef]{1,16})(?:-([0-9abcdef]{1,16}))?$", parameter.replace("0x", "").lower())
+    if not m:
+        raise ValueError(f"{parameter}: invalid range specified: not a number")
+    # check that start < end
+    start = min(int(m.group(1).replace("0x", ""), 16), int(m.group(2).replace("0x", ""), 16))
+    end = max(int(m.group(1).replace("0x", ""), 16), int(m.group(2).replace("0x", ""), 16)) or start
+    if start > end:
+        raise ValueError(f"{parameter}: invalid range specified: start > end")
+    return [start, end]
+
+def cast_expand_path(parameter: Any) -> Optional[str]:
+    if parameter is None:
+        return None
+    exp_str = os.path.expandvars(parameter)
+    # ensure exists
+    p = Path(exp_str)
+    if not p.exists():
+        raise FileNotFoundError(f"Path {p} doesn't exist")
+    # return string and not PosixPath, since this object is not serializable
+    return str(p)
+
+def cast_expand_path_no_verify(parameter: Any) -> Optional[str]:
+    if parameter is None:
+        return None
+    exp_str = os.path.expandvars(parameter)
+    return exp_str
+
+# register validators
+settings.validators.register(
+    # general
+    Validator("workdir", must_exist=True, cast=cast_expand_path_no_verify),
+    Validator("work_dir", must_exist=True, default=lambda config, _validator: config.workdir),
```

Comment:
Happy to revert this and just use KAFL_WORK_DIR. Changing --work-dir or keeping the inconsistency are also not great options, and the semantics/usage of this variable is changing now anyway (used by default).